### ISSUE TITLE
Plugin parsing error when type mismatch

### ISSUE THREAD

**Issue by ipedrazas** [drone-plugins/drone-docker#11]
```
panic: Unable to unarmshal vargs. json: cannot unmarshal number into Go value of type string

goroutine 1 [running]:
github.com/drone/drone-plugin-go/plugin.MustParse()
	/drone/src/github.com/drone/drone-plugin-go/plugin/param.go:129 +0x59
main.main()
	/drone/src/github.com/drone-plugins/drone-docker/main.go:37 +0x3fc
[debug] Running post-Build Cache steps
```

**Comment by bradrydzewski** [comment]
@ipedrazas if the plugin expects a string, and $BUILD_NUMBER is a number, this will cause the plugin to fail parsing input and will error. You'll need to quote the value in your yaml. `"$$BUILD_NUMBER"`

**Comment by bradrydzewski** [comment]
unfortunately due to the static nature of Go, there is not really any way around this, aside from a friendlier error message

**Comment by ipedrazas** [comment]
Sure, but we should put this in some doc. I mean, I know that it's expecting a string, but using the $BUILD_NUMBER is common practice

**Comment by ipedrazas** [comment]
It's not all about the code :) I'm happy to send a PR with some improved doc.

**Comment by bradrydzewski** [comment]
it is in the docs in reference to the docker tag field :)

> Note that in the above example we quote the version numbers. If the yaml parser interprets the value as a number it will cause a parsing error.

**Comment by ipedrazas** [comment]
:facepalm: :)

**Comment by bradrydzewski** [comment]
but yes, if there is a way to make more clear definitely send a pull request !!  some things seem obvious to me (because I stare at them all day) and may not jump out to the reader

**Comment by ipedrazas** [comment]
:+1:

**Comment by bradrydzewski** [comment]
I'm thinking we can instead use custom error messaging in our plugin parser. We can intercept and cast the JSON error to print out an error message that is meaningful to Drone, like this:
http://play.golang.org/p/nCDGxhW41X

**Comment by ipedrazas** [comment]
The special type sounds like the way to go. Truth is that tags in images are strings, but if it can handle both situations... it feels good.

**Comment by msteinert** [comment]
This appears to work too...

http://play.golang.org/p/qnbOQQpCeW

**Comment by bradrydzewski** [comment]
this is resolved in 0.5
